tests/parametric/test_config_consistency.py

```diff
@@ -152,3 +152,31 @@ def test_dd_trace_agent_http_url_nonexistent(self, library_env, test_agent, test
         assert resp["dd_trace_agent_url"] == "http://random-host:9999/"
         with pytest.raises(ValueError):
             test_agent.wait_for_num_traces(num=1, clear=True)
+
+
+@scenarios.parametric
+@features.tracing_configuration_consistency
+class Test_Config_RateLimit:
+    # The default value of DD_TRACE_RATE_LIMIT is validated using the tracer configuration.
+    # This approach avoids the need to create a new weblog endpoint that generates 100 traces per second,
+    # which would be unreliable for testing and require significant effort for each tracer's weblog application.
+    # The feature is mainly tested in the second test case, where the rate limit is set to 1 to ensure it works as expected.
+    @parametrize("library_env", [{}])
+    def test_default_trace_rate_limit(self, library_env, test_agent, test_library):
+        with test_library as t:
+            resp = t.get_tracer_config()
+        assert resp["dd_trace_rate_limit"] == "100"
+
+    @parametrize("library_env", [{"DD_TRACE_RATE_LIMIT": "1"}])
+    def test_setting_trace_rate_limit(self, library_env, test_agent, test_library):
+        with test_library:
+            with test_library.start_span(name="s1") as s1:
+                pass
+            with test_library.start_span(name="s2") as s2:
+                pass
```

Comment:
Is it safe to assume these spans will both be generated in <1s?

Reply:
Same response as in the thread:

"Yeah so it might be flaky? i guess we'll see how other libraries handle it. Seems to be fine for Python, but I don't really know how else to test it since the accepted input for the variable is a positive integer and 1 is the lowest accepted value"